service/src/main/java/org/cbioportal/genome_nexus/service/annotation/VariantClassificationResolver.java

```diff
@@ -0,0 +1,163 @@
+package org.cbioportal.genome_nexus.service.annotation;
+
+import org.cbioportal.genome_nexus.model.TranscriptConsequence;
+import org.cbioportal.genome_nexus.model.VariantAnnotation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Component
+public class VariantClassificationResolver
+{
+    public static Map<String, String> VARIANT_MAP = initVariantMap();
+
+    private final CanonicalTranscriptResolver canonicalTranscriptResolver;
+    private final TranscriptConsequencePrioritizer consequencePrioritizer;
+    private final VariantTypeResolver variantTypeResolver;
+
+    @Autowired
+    public VariantClassificationResolver(CanonicalTranscriptResolver canonicalTranscriptResolver,
+                                         VariantTypeResolver variantTypeResolver,
+                                         TranscriptConsequencePrioritizer consequencePrioritizer)
+    {
+        this.canonicalTranscriptResolver = canonicalTranscriptResolver;
+        this.consequencePrioritizer = consequencePrioritizer;
+        this.variantTypeResolver = variantTypeResolver;
+    }
+
+    public String resolve(VariantAnnotation variantAnnotation)
+    {
+        TranscriptConsequence canonicalTranscript = this.canonicalTranscriptResolver.resolve(variantAnnotation);
+
+        // resolve for the canonical transcript
+        // if canonical transcript resolver returns null, then most severe consequence will be used
+        return this.resolve(variantAnnotation, canonicalTranscript);
+    }
+
+    public String resolve(VariantAnnotation variantAnnotation, TranscriptConsequence transcriptConsequence)
+    {
+        String variantClassification = null;
+        String variantType = this.variantTypeResolver.resolve(variantAnnotation);
+        Boolean isInframe = this.isInframe(variantAnnotation);
+
+        // if not null, try to resolve the variant classification for the provided transcript
+        if (transcriptConsequence != null)
+        {
+            variantClassification = this.resolveVariantClassification(
+                this.consequencePrioritizer.pickHighestPriorityConsequence(transcriptConsequence.getConsequenceTerms()),
```

Comment:
@jjgao @inodb this logic is adopted from the annotation pipeline repository, and it only resolves to a single mutation type. I think this is what we want when annotating MAF files. But for hotspot filtering purposes should we consider all consequence terms and return a list of mutation types instead of just one result?

Reply:
Yeah after merging #132 we should be able to match on hugo symbol and consequence for any of the transcripts. If one of them matches it's a hotspot